src/test/java/seedu/address/logic/parser/ParserUtilTest.java

```diff
@@ -55,154 +39,74 @@ public void parseIndex_validInput_success() throws Exception {
-    public void parseTempRange_validValueWithWhitespace_returnsTrimmedTempRange() throws Exception {
-        String tempRangeWithWhitespace = WHITESPACE + VALID_TEMP_RANGE + WHITESPACE;
-        TemperatureRange expectedTempRange = new TemperatureRange(VALID_TEMP_RANGE);
-        assertEquals(expectedTempRange, ParserUtil.parseTemperatureRange(tempRangeWithWhitespace));
-    }
-
-    @Test
-    public void parseTemperature_invalidValue_throwsParseException() {
-        assertThrows(ParseException.class, () -> ParserUtil.parseTemperature(INVALID_TEMPERATURE));
-    }
-
-    @Test
-    public void parseTemperature_validValueWithoutWhitespace_returnsTemperature() throws Exception {
-        Temperature expectedTemperature = new Temperature(VALID_TEMPERATURE);
-        assertEquals(expectedTemperature, ParserUtil.parseTemperature(VALID_TEMPERATURE));
+    public void parsePositiveInteger_validInput_success() throws Exception {
+        // No whitespaces
+        assertEquals(1, ParserUtil.parsePositiveInteger("1"));
+        // Leading and trailing whitespaces
+        assertEquals(1, ParserUtil.parsePositiveInteger("  1  "));
     }
 
     @Test
-    public void parseTemperature_validValueWithWhitespace_returnsTrimmedTemperature() throws Exception {
-        String tempWithWhitespace = WHITESPACE + VALID_TEMPERATURE + WHITESPACE;
-        Temperature expectedTemperature = new Temperature(VALID_TEMPERATURE);
-        assertEquals(expectedTemperature, ParserUtil.parseTemperature(tempWithWhitespace));
-    }
+    public void arePrefixesPresent() {
+        Prefix prefixOnePresent = mock(Prefix.class);
+        Prefix prefixTwoPresent = mock(Prefix.class);
+        Prefix prefixThreeMissing = mock(Prefix.class);
 
-    @Test
-    public void parseTemperature_null_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> ParserUtil.parseTemperature((String) null));
-    }
+        ArgumentMultimap argMultimap = mock(ArgumentMultimap.class);
+        when(argMultimap.getValue(prefixOnePresent)).thenReturn(Optional.of("one"));
+        when(argMultimap.getValue(prefixTwoPresent)).thenReturn(Optional.of("two"));
+        when(argMultimap.getValue(prefixThreeMissing)).thenReturn(Optional.empty());
 
-    @Test
-    public void parsePhone_null_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> ParserUtil.parsePhone((String) null));
-    }
+        // One missing prefix
+        assertFalse(ParserUtil.arePrefixesPresent(argMultimap, prefixThreeMissing));
+        // One present and one missing prefixes
+        assertFalse(ParserUtil.arePrefixesPresent(argMultimap, prefixOnePresent, prefixThreeMissing));
 
-    @Test
-    public void parsePhone_invalidValue_throwsParseException() {
-        assertThrows(ParseException.class, () -> ParserUtil.parsePhone(INVALID_PHONE));
+        // No prefixes
+        assertTrue(ParserUtil.arePrefixesPresent(argMultimap));
+        // Two present prefixes
```

Comment:
Maybe can include space between method and javaDoc to see better

Reply:
Ya agree with you on this